be/src/storage/rowset/bitmap_index_writer.cpp

```diff
@@ -115,14 +116,64 @@ class BitmapUpdateContext {
     bool _size_changed{false};
 };
 
+// if last bit is 0 it is std::unique_ptr<BitmapUpdateContext>
+// else it is a single value
+class BitmapUpdateContextRefOrSingleValue {
```

Comment:
Did you consider replacing Roaring with BitmapValue in BitmapUpdateContext?
I think BitmapValue is more memory efficient than Roaring because it will only allocate it when it is needed.
BitmapValue may be more efficient than this CL solution when the distinct values are more than 1.
In this CL, if the distinct values are more than 1, it will be the same as before. But with BitmapValue, it will use less memory.

Reply:
> Did you consider replacing Roaring with BitmapValue in BitmapUpdateContext?
Tried this today, just found out they have different serialization format and is not compatible with old format, are you sure to continue this path?